cpu/atmega2560/periph/gpio.c

```diff
@@ -86,51 +86,36 @@ static inline uint16_t _pin_addr(gpio_t pin)
     return (_port_addr(pin) - 0x02);
 }
 
-int gpio_init(gpio_t pin, gpio_dir_t dir, gpio_pp_t pullup)
+int gpio_init(gpio_t pin, gpio_mode_t mode)
 {
-    int res;
-
-    if (dir == GPIO_DIR_OUT) {
+    if (mode == GPIO_OUT) {
         _SFR_MEM8(_ddr_addr(pin)) |= (1 << _pin_num(pin));
-        res = bit_is_set(_SFR_MEM8(_ddr_addr(pin)), _pin_num(pin));
```

Comment:
Nope, there is no reason for this check what so ever, don't ask how this slipped through review...